rts/motoko-rts/src/allocator.rs

```diff
@@ -0,0 +1,43 @@
+// c.f. https://os.phil-opp.com/heap-allocation/#dynamic-memory
+
+use alloc::alloc::{GlobalAlloc, Layout};
+//use core::ptr::null_mut;
+use crate::memory::{alloc_blob, ic};
+use crate::types::Bytes;
+
+pub struct HeapAllocator;
+
+unsafe impl GlobalAlloc for HeapAllocator {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
```

Comment:
Maybe important to document: The allocator is only used for temporary objects whose pointers are not retained across GC increments...

Reply:
Absolutely. Will do

Reply:
Done, also renamed to EphemeralAlloctor